File: bridge/python/tests/edit.py
from destack import Edit

from session import open_workspace


def test_edit_source() -> None:
    fixture = open_workspace(
        [
            ("destack.json", '{"name":"@test/app"}'),
            ("src/index.ds", "export const value = 1;"),
        ],
    )
    workspace = fixture.workspace

    commit = workspace.edit(
        [
            Edit.set_text("src/next.ds", "export const next = 2;"),
        ],
    )

    assert commit.before.id != commit.after.id
    assert [file.path for file in workspace.files()] == [
        "destack.json",
        "src/index.ds",
        "src/next.ds",
    ]

File: bridge/rust/tests/edit.rs
#[path = "support/session.rs"]
mod session;

use session::open_workspace;

#[test]
fn test_edit_source() -> destack::Result<()> {
    let fixture = open_workspace(&[
        ("destack.json", r#"{"name":"@test/app"}"#),
        ("src/index.ds", "export const value = 1;"),
    ])?;
    let workspace = fixture.workspace;

    // apply one edit
    let commit = workspace.edit(vec![destack::Edit::set_text(
        "src/next.ds",
        "export const next = 2;",
    )])?;

    // read files after the edit
    let paths = workspace
        .files()?
        .into_iter()
        .map(|file| file.path)
        .collect::<Vec<_>>();

    assert_ne!(commit.before.id, commit.after.id);
    assert_eq!(paths, ["destack.json", "src/index.ds", "src/next.ds"]);

    Ok(())
}
